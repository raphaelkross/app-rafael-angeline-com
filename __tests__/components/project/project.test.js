/* eslint-env jest */

import renderer from "react-test-renderer";
import Project from "../../../components/project/project";

describe("Project", () => {
	it("should render complete layout", () => {
		const content =
			"<h4>Above Title</h4><h3>main Title</h3><p>Main content</p>";

		const summary = {
			title: "Rafael Angeline",
			date: "June 26th, 2018",
			stack: "WP, React",
			link: "https://github.com"
		};

		const component = renderer.create(
			<Project
				title="Project Title Tag"
				content={content}
				summary={summary}
			/>
		);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
