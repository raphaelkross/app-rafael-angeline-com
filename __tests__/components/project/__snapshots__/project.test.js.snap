// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should render complete layout 1`] = `
Array [
  <header
    className="jsx-2352544197"
  >
    <div
      className="jsx-2352544197 container"
    >
      <nav
        className="jsx-2352544197"
      >
        <ul>
          <li>
            <a
              href="http://localhost:3000/"
              onClick={[Function]}
            >
              <span>
                Home.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/about"
              onClick={[Function]}
            >
              <span>
                About.
              </span>
            </a>
            <ul>
              <li>
                <a
                  href="https://twitter.com/RafaelAngeline"
                  onClick={[Function]}
                >
                  <span>
                    Twitter.
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/raphaelkross"
                  onClick={[Function]}
                >
                  <span>
                    Github.
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="https://profiles.wordpress.org/rafaelangeline"
                  onClick={[Function]}
                >
                  <span>
                    WordPress.
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li>
            <a
              href="http://localhost:3000/work"
              onClick={[Function]}
            >
              <span>
                Work.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/contact"
              onClick={[Function]}
            >
              <span>
                Contact.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/blog"
              onClick={[Function]}
            >
              <span>
                Blog.
              </span>
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="jsx-2352544197 responsive-trigger"
      >
        <a
          className="jsx-2352544197 responsive-trigger-link"
          href="#"
          onClick={[Function]}
        >
          <svg
            className="jsx-2352544197 feather feather-menu"
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              className="jsx-2352544197"
              x1="3"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              className="jsx-2352544197"
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              className="jsx-2352544197"
              x1="3"
              x2="21"
              y1="18"
              y2="18"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="jsx-2352544197 modal"
    >
      <a
        className="jsx-2352544197 modal-close"
        href="#"
        onClick={[Function]}
      >
        <svg
          className="jsx-2352544197 feather feather-x"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            className="jsx-2352544197"
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            className="jsx-2352544197"
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </a>
      <div
        className="jsx-2352544197"
      >
        <ul>
          <li>
            <a
              href="http://localhost:3000/"
              onClick={[Function]}
            >
              <span>
                Home.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/about"
              onClick={[Function]}
            >
              <span>
                About.
              </span>
            </a>
            <ul>
              <li>
                <a
                  href="https://twitter.com/RafaelAngeline"
                  onClick={[Function]}
                >
                  <span>
                    Twitter.
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/raphaelkross"
                  onClick={[Function]}
                >
                  <span>
                    Github.
                  </span>
                </a>
              </li>
              <li>
                <a
                  href="https://profiles.wordpress.org/rafaelangeline"
                  onClick={[Function]}
                >
                  <span>
                    WordPress.
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li>
            <a
              href="http://localhost:3000/work"
              onClick={[Function]}
            >
              <span>
                Work.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/contact"
              onClick={[Function]}
            >
              <span>
                Contact.
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/blog"
              onClick={[Function]}
            >
              <span>
                Blog.
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </header>,
  <main
    className="jsx-2760187776"
  >
    <div
      className="jsx-2760187776 container"
    >
      <div
        className="jsx-2402135458 content project-details"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h4>Above Title</h4><h3>main Title</h3><p>Main content</p>",
          }
        }
      />
      <aside
        className="jsx-1339113702 project-summary"
      >
        <label
          className="jsx-1339113702"
          for="project-title"
        >
          Project
        </label>
        <h1
          className="jsx-1339113702"
          id="project-title"
        >
          Rafael Angeline
        </h1>
        <dl
          className="jsx-1339113702"
        >
          <dt
            className="jsx-1339113702"
          >
            Date:
          </dt>
          <dd
            className="jsx-1339113702 date"
          >
            June 26th, 2018
          </dd>
          <dt
            className="jsx-1339113702"
          >
            Stack:
          </dt>
          <dd
            className="jsx-1339113702 stack"
          >
            WP, React
          </dd>
        </dl>
        <a
          className="jsx-1339113702 btn"
          href="https://github.com"
          target="_blank"
        >
          Check the Project
        </a>
      </aside>
    </div>
  </main>,
  <footer
    className="jsx-23269780"
  >
    <div
      className="jsx-23269780 container"
    >
      <div
        className="jsx-23269780 widget"
      >
        <h3
          className="jsx-23269780"
        >
          Contact.
        </h3>
        <p
          className="jsx-23269780"
        >
          Rafael Angeline
          <br
            className="jsx-23269780"
          />
          Send me your message
          <br
            className="jsx-23269780"
          />
          me [at] rafaelangeline.com
        </p>
      </div>
      <div
        className="jsx-23269780 widget"
      >
        <h3
          className="jsx-23269780"
        >
          Location.
        </h3>
        <p
          className="jsx-23269780"
        >
          Magic Happens at
          <br
            className="jsx-23269780"
          />
          SÃ£o Paulo, SP - Brazil
        </p>
      </div>
      <div
        className="jsx-23269780 widget"
      >
        <h3
          className="jsx-23269780"
        >
          Stay in Touch.
        </h3>
        <p
          className="jsx-23269780"
        >
          Follow me on
           
          <a
            className="jsx-23269780 twitter"
            href="https://twitter.com/RafaelAngeline"
            target="_blank"
          >
            <span
              className="jsx-23269780"
            >
              Twitter
            </span>
          </a>
          ,
           
          <a
            className="jsx-23269780 stackoverflow"
            href="http://stackexchange.com/users/8383862/rafael-angeline"
            target="_blank"
          >
            <span
              className="jsx-23269780"
            >
              StackOverflow
            </span>
          </a>
          ,
           
          <a
            className="jsx-23269780 github"
            href="https://github.com/raphaelkross"
            target="_blank"
          >
            <span
              className="jsx-23269780"
            >
              Github
            </span>
          </a>
          ,
           
          <a
            className="jsx-23269780 wordpress"
            href="https://profiles.wordpress.org/rafaelangeline"
            target="_blank"
          >
            <span
              className="jsx-23269780"
            >
              WordPress
            </span>
          </a>
        </p>
      </div>
      <p
        className="jsx-23269780 copy"
      >
        Â© Copyright Rafael Angeline. Built by Rafael Angeline with React, Next.js and WordPress &lt;3
      </p>
    </div>
  </footer>,
]
`;
